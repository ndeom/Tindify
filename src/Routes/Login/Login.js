import React from "react";
import "./Login.scss";

export default function Login() {
  return (
    <div id="login" data-testid="login">
      <h1 data-testid="login-title">Tindify</h1>
      <a
        href="https://tindify-web.herokuapp.com/api/login"
        data-testid="login-button"
      >
        {" "}
        Sign in with Spotify{" "}
      </a>
      {/* <a href="http://localhost:8080/api/login"> Sign in with Spotify </a> */}
    </div>
  );
}
