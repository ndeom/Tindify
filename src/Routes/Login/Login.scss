@import url("https://fonts.googleapis.com/css2?family=Nunito+Sans:wght@400;600;700;800;900&display=swap");

$light-black: #181818;
$green-light-back: #1aa34a;
$green-dark-back: #1db954;

#login {
  //background-color: $light-black;
  height: 100vh;
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  font-family: "Nunito Sans";

  h1 {
    font-weight: 900;
    font-size: 70px;
    color: white;
    margin-bottom: 100px;
  }

  a {
    font-weight: 700;
    font-size: 18px;
    color: white;
    background-color: $green-dark-back;
    text-decoration: none;
    line-height: 1;
    padding: 1rem 2rem;
    border-radius: 999px;
    letter-spacing: 1px;
    min-width: 160px;
    transition: filter 150ms cubic-bezier(0.215, 0.61, 0.355, 1);

    &:hover {
      filter: brightness(1.1);
    }
  }
}

@media only screen and (max-width: 500px) {
  #login {
    h1 {
      margin-bottom: 50px;
    }
  }
}
