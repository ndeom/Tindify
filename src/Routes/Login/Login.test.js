import React from "react";
import { render } from "@testing-library/react";
import TestRenderer from "react-test-renderer";
import Login from "./Login";

it("should contain an h1 element and an anchor button", () => {
  const { getByTestId } = render(<Login />);

  expect(getByTestId("login-title")).toBeInTheDocument();
  expect(getByTestId("login-title")).toHaveTextContent("Tindify");

  expect(getByTestId("login-button")).toBeInTheDocument();
  expect(getByTestId("login-button")).toHaveTextContent("Sign in with Spotify");
});

it("should match the snapshot", () => {
  const tree = TestRenderer.create(<Login />);
  expect(tree).toMatchSnapshot();
});
