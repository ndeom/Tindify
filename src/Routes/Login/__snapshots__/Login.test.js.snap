// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
<div
  data-testid="login"
  id="login"
>
  <h1
    data-testid="login-title"
  >
    Tindify
  </h1>
  <a
    data-testid="login-button"
    href="https://tindify-web.herokuapp.com/api/login"
  >
     
    Sign in with Spotify
     
  </a>
</div>
`;
